### PR TITLE
Fix/opt out of rendering

### PR DESCRIPTION
**Pull request by bkapustik**
# Motivation

Disable default rendering of tags in the Xperience administration pagebuilder and preview.
Add option to select rendering in pagebuilder or preview or both.

## Checklist

- [x] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [x] Tests are passing
- [x] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

## How to test

If manual testing is required, what are the steps?

### CODE REVIEW

File: src/Kentico.Xperience.TagManager/Admin/CodeSnippetConfigurationModel.cs [hunk 1]
```diff
@@ -47,13 +50,19 @@ public void MapToChannelCodeSnippetInfo(ChannelCodeSnippetItemInfo info)
...
-internal static class CodeSnippetLocationsExtensions
+internal static class CodeSnippetExtensions
 {
-    public const string FormComponentOptions = $"{nameof(CodeSnippetLocations.HeadTop)};Insert at the top of the head\r\n" +
+    public const string LocationFormComponentOptions = $"{nameof(CodeSnippetLocations.HeadTop)};Insert at the top of the head\r\n" +
                                                $"{nameof(CodeSnippetLocations.HeadBottom)};Insert at the bottom of the head\r\n" +
                                                $"{nameof(CodeSnippetLocations.BodyTop)};Insert at the top of the body\r\n" +
                                                $"{nameof(CodeSnippetLocations.BodyBottom)};Insert at the bottom of the body\r\n";
+
+    public const string DisplayModeFormComponentOptions = $"{nameof(CodeSnippetAdministrationDisplayMode.None)};Do not dispaly in Administration\r\n" +
```
Comment (seangwright) [inline src/Kentico.Xperience.TagManager/Admin/CodeSnippetConfigurationModel.cs:64]:
```suggestion
    public const string DisplayModeFormComponentOptions = $"{nameof(CodeSnippetAdministrationDisplayMode.None)};Do not display in Administration\r\n" +
```

File: src/Kentico.Xperience.TagManager/Admin/CodeSnippetConfigurationModel.cs [hunk 2]
```diff
@@ -47,13 +50,19 @@ public void MapToChannelCodeSnippetInfo(ChannelCodeSnippetItemInfo info)
...
+                                                $"{nameof(CodeSnippetAdministrationDisplayMode.PreviewOnly)};Display in the Preview only\r\n" +
```
Comment (seangwright) [inline src/Kentico.Xperience.TagManager/Admin/CodeSnippetConfigurationModel.cs:65]:
```suggestion
                                                $"{nameof(CodeSnippetAdministrationDisplayMode.PreviewOnly)};Display in the Preview view mode only\r\n" +
```

File: src/Kentico.Xperience.TagManager/Admin/CodeSnippetConfigurationModel.cs [hunk 3]
```diff
@@ -47,13 +50,19 @@ public void MapToChannelCodeSnippetInfo(ChannelCodeSnippetItemInfo info)
...
+                                                $"{nameof(CodeSnippetAdministrationDisplayMode.PageBuilderOnly)};Display in the PageBuilder only\r\n" +
```
Comment (seangwright) [inline src/Kentico.Xperience.TagManager/Admin/CodeSnippetConfigurationModel.cs:66]:
```suggestion
                                                $"{nameof(CodeSnippetAdministrationDisplayMode.PageBuilderOnly)};Display in the Page Builder only\r\n" +
```

File: src/Kentico.Xperience.TagManager/Admin/CodeSnippetConfigurationModel.cs [hunk 4]
```diff
@@ -47,13 +50,19 @@ public void MapToChannelCodeSnippetInfo(ChannelCodeSnippetItemInfo info)
...
+                                                $"{nameof(CodeSnippetAdministrationDisplayMode.Both)};Display in Preview and Pagebuilder";
```
Comment (seangwright) [inline src/Kentico.Xperience.TagManager/Admin/CodeSnippetConfigurationModel.cs:67]:
```suggestion
                                                $"{nameof(CodeSnippetAdministrationDisplayMode.Both)};Display in the Preview view mode and the Page Builder";
```

**Review by seangwright — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -102,6 +102,7 @@ dotnet add package Kentico.Xperience.TagManager
        - Top of the body - inserts a script immediately after the opening body tag.
        - Bottom of the body - inserts a script right before the closing body tag.
    - Fill in a consent if required.
+   - Select whether you want to display tags in the Xperience administration preview or pagebuilder, or both. This option defaults to None.
```
Comment (michalJakubis) [inline README.md:105]:
```suggestion
   - Select whether you want to display tags in the Xperience administration preview or Page Builder, or both. This option defaults to None.
```

File: src/Kentico.Xperience.TagManager/Rendering/DefaultChannelCodeSnippetsService.cs [hunk 6]
```diff
@@ -99,13 +101,18 @@ private static IEnumerable<CodeSnippetDto> CreateCodeSnippet(ChannelCodeSnippetI
 
         var tags = new List<CodeSnippetDto>();
 
+        if (!Enum.TryParse(snippetInfo.ChannelCodeSnippetAdministrationDisplayMode, out CodeSnippetAdministrationDisplayMode displayMode))
+        {
+            displayMode = CodeSnippetAdministrationDisplayMode.None;
+        }
```
Comment (michalJakubis) [inline src/Kentico.Xperience.TagManager/Rendering/DefaultChannelCodeSnippetsService.cs:107]: Add blank line after a closing brace.

**Comment by seangwright** [comment]
@bkapustik it looks like all the suggestions I requested were reverted in [the most recent commit](https://github.com/Kentico/xperience-by-kentico-tag-manager/pull/22/commits/d25c1af8508a081d05a459466720b98da3b78a7e). Can you fix this?

**Review by seangwright — APPROVED** [review]

**Pull request merged**